### PR TITLE
wildcard prefix hostname validate fix

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fixes: #12720

This kind of `default/productpage` is invalid

### CODE REVIEW

File: pilot/pkg/model/validation.go [hunk 1]
```diff
@@ -39,9 +39,9 @@ import (
 
 const (
 	dns1123LabelMaxLength int    = 63
-	dns1123LabelFmt       string = "[a-zA-Z0-9]([-a-z-A-Z0-9]*[a-zA-Z0-9])?"
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/validation.go]: This is a bug

Comment (rshriram) [reply pilot/pkg/model/validation.go]: why? caps is allowed in dns labels right?

File: pilot/pkg/model/validation.go [hunk 2]
```diff
@@ -39,9 +39,9 @@ import (
...
+	dns1123LabelFmt       string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
 	// a wild-card prefix is an '*', a normal DNS1123 label with a leading '*' or '*-', or a normal DNS1123 label
-	wildcardPrefix string = `\*|(\*|\*-)?(` + dns1123LabelFmt + `)`
+	wildcardPrefix string = `(\*|(\*|\*-))?(` + dns1123LabelFmt + `)`
```
Comment (rshriram) [inline pilot/pkg/model/validation.go]: why change this? its currently * or *dns1123label..
what you have is
(* or * or *-)dns1123label

File: pilot/pkg/model/validation.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/model/validation.go]: No, A-Z is not allowed

File: pilot/pkg/model/validation.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/model/validation.go]: The previous regex can match `a/b/c` and many other invalid format

File: pilot/pkg/model/validation.go [hunk 3]
```diff
@@ -299,6 +303,9 @@ func IsDNS1123Label(value string) bool {
 // IsWildcardDNS1123Label tests for a string that conforms to the definition of a label in DNS (RFC 1123), but allows
 // the wildcard label (`*`), and typical labels with a leading astrisk instead of alphabetic character (e.g. "*-foo")
 func IsWildcardDNS1123Label(value string) bool {
+	if value == "*" {
+		return true
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/validation.go]: I make "*" match separately

**Pull request closed without merging**